pkg/resources/cruisecontrol/configmap.go

```diff
@@ -162,30 +159,50 @@ func GenerateCapacityConfig(kafkaCluster *v1beta1.KafkaCluster, log logr.Logger,
 
 	capacityConfig := CapacityConfig{}
 
-	for _, broker := range kafkaCluster.Spec.Brokers {
-		brokerDisks, err := generateBrokerDisks(broker, kafkaCluster.Spec, log)
-		if err != nil {
-			return "", errors.WrapIfWithDetails(err, "could not generate broker disks config for broker", "brokerID", broker.Id)
-		}
-		brokerCapacity := BrokerCapacity{
-			BrokerID: fmt.Sprintf("%d", broker.Id),
-			Capacity: Capacity{
-				DISK:  brokerDisks,
-				CPU:   generateBrokerCPU(broker, kafkaCluster.Spec, log),
-				NWIN:  generateBrokerNetworkIn(broker, kafkaCluster.Spec, log),
-				NWOUT: generateBrokerNetworkOut(broker, kafkaCluster.Spec, log),
-			},
-			Doc: defaultDoc,
+	var strkeys []string
```

Comment:
This can be pre initialized using make.
```suggestion
	strkeys := make([]string, 0, len(kafkaCluster.Status.BrokersState))
```

Reply:
This variable should be called something like brokerIdsFromStatus

Reply:
Yes, I should rename it, will do that, but if I pre initilaize it than under that when appending stuff to it I would instead have to have an index and use that in the for cycle. This seemed easier and a lot cleaner to me really.

Reply:
Yes you are right, that is why I used the 0 as the middle parameter for make. When you are specifying that you are safe to use append, since it will append all the data to the preallocated slice. If you omitting that you will end up with a slice where you have some (in case of a string slice) empty strings in the beginning.